### DISCUSSION TITLE
Hints to quarto of where to look for a jupyter executable?

### DISCUSSION THREAD

**Discussion by dmbates** [quarto-dev/quarto-cli#1540] [Q&A] [answered]
My colleagues and I are very much enjoying using Quarto for literate programming with Julia (and looking forward to this [Juliacon presentation](https://pretalx.com/juliacon-2022/talk/KKBXAZ/) by @jjallaire on Friday).

If a Julia user doesn't have jupyter installed when they add the [IJulia.jl](https://github.com/JuliaLang/IJulia.jl) package a Conda instance (currently the miniforge flavor IIRC) will get installed in their `~/.julia` folder (Mac and Linux - not sure where it is under Windows) with the jupyter executable at `~/.julia/conda/3/bin/jupyter`.

What would be a good way of having this location recognized as `jupyter` in the quarto configuration?  For myself I just create a symbolic link from my `~/bin` directory, which is on my PATH, to this executable and everything is cool.  But I am writing [instructions](https://RePsychLing.github.io/SMLP2022) for students in a short course in September and they may not be the sort who are thrilled when asked to "create a symbolic link"

**Comment by tpoisot** [comment]
I second this entirely! If possible, it would make sense to use the jupyter version that is bundled with *IJulia.jl*, rather than having multiple installations on the same system.

**Comment by jjallaire** [comment]
You can define a `QUARTO_PYTHON` environment variable which will override our internal checking of PATH, etc. for the version of Python to use.

**Reply by dmbates** [reply]
@jjallaire I can check this, of course, but is there a similar environment variable for jupyter?  My concern is with the evaluation of Julia code blocks after a .qmd file is converted to a Jupyter notebook.  Is it assumed that the jupyter executable will be in the same directory as the python executable?

**Reply by jjallaire** [reply]
We actually don't invoke Jupyter via the executable (which is no longer recommended due to PATH pollution created by the many versions of Python in play on a system). Instead we do the following when discovering kernels:

```bash
python3 -m jupyter --paths --json
```

That's the only time we actually invoke the `jupyter` module/executable. All of the rest of our interaction is via the Python packages `nbformat`, `nbclient`, etc.

So `QUARTO_PYTHON` is effectively `QUARTO_JUPYTER` in Quarto so might e.g. resolve to this command:

```bash
/Users/jjallaire/.julia/conda/3/bin/python3 -m jupyter --paths --json
```

**Reply by dmbates** [reply]
Great!  Thanks for the clarification.

**Comment by jjallaire** [comment, marked as answer]
I have just made a change to have Quarto look specifically for the IJulia installed Conda whenever we are using a Julia kernel. This is available in our v1.1.77 pre-release: https://quarto.org/docs/download/prerelease.html. 

This should mean that users don't need to worry about having a separate Jupyter installation and should also be more reliable as the IJulia bundled kernel seems less likely to have installation/configuration vagaries that could muck things up.
